File: Python/ProjectEuler3.py
import math

def findFactor(number):
    upperBound = int(math.sqrt(number)) + 1 
    return [x for x in range(2,upperBound) if number % x == 0]

def isPrime(number): 
    return  len(findFactor(number)) == 0

def filterList(number):
    upperBound = int(math.sqrt(number)) + 1
    list = range(2,upperBound)
    return max(filter(lambda x: isPrime(x),filter(lambda y: number % y == 0,list)))

print(filterList(600851475143))

File: FSharp/ProjectEuler3.fs
open System 

let findFactor (number:int64) = 
    let upperBound = int64(sqrt (double number))
    [2L..upperBound] |> Seq.filter(fun x -> number % x = 0L)

let isPrime(n:int64) =  (findFactor(n) |> Seq.length) = 0

let filterList (number:int64) =
        let upperBound = int64(sqrt (double number))
        [2L..upperBound] |> Seq.filter(fun x -> number % x = 0L) |> Seq.filter(fun x -> isPrime(x)) |> Seq.max

[<EntryPoint>]
let main argv =
    let result = filterList(600851475143L)
    printfn "%A" result
    0 // return an integer exit code